src/ranged.cpp

```diff
@@ -1558,9 +1558,9 @@ static void cycle_action( item &weap, const tripoint &pos )
 
     if( weap.ammo_data() && weap.ammo_data()->ammo->casing ) {
         const itype_id casing = *weap.ammo_data()->ammo->casing;
-        if( weap.has_flag( "RELOAD_EJECT" ) || weap.gunmod_find( itype_brass_catcher ) ) {
-            weap.put_in( item( casing ).set_flag( "CASING" ), item_pocket::pocket_type::CONTAINER );
-        } else {
+        item *brass_catcher = weap.gunmod_find( itype_brass_catcher );
+        if( !( weap.has_flag( "RELOAD_EJECT" ) || ( brass_catcher && brass_catcher->put_in( item( casing ),
+                item_pocket::pocket_type::CONTAINER ).success() ) ) ) {
```

Comment:
I'm not sure if this logic is correct. If a weapon has the "RELOAD_EJECT" flag, before this change it would put the casing in the weapon, but after this it will do nothing, making the casing disappear.

Also, later at line 1579-1581 ammo linkages are also added to brass catchers, which should also be updated.

Reply:
Oops, I guess I confused myself with prior changes to that. Not that putting casings into the weapon would have worked before, though. Weapons don't have any container pocket, so it just did nothing. I'll separate the "RELOAD_EJECT" logic and leave it as it was before for another fix.